File: 1310.子数组异或查询/solution.cpp
class Solution {
public:
    vector<int> xorQueries(vector<int>& arr, vector<vector<int>>& queries) {
        int sz=arr.size();
        vector<int> vec(sz+1);
        for (int i=0;i<sz;i++) {
            vec.at(i+1)=vec.at(i)^arr.at(i);
        }
        int qSz=queries.size();
        vector<int> ans(qSz);
        for (int i=0;i<qSz;i++) {
            int a=queries.at(i).at(0),b=queries.at(i).at(1);
            ans.at(i)=vec.at(b+1)^vec.at(a);
        }
        return ans;
    }
};

File: 1310.子数组异或查询/solution.go
func xorQueries(arr []int, queries [][]int) []int {
    szA,szQ:=len(arr),len(queries)
    vec:=make([]int,szA+1)
    for i:=0;i<szA;i++ {
        vec[i+1]=vec[i]^arr[i]
    }
    ans:=make([]int,szQ)
    for i:=0;i<szQ;i++ {
        a,b:=queries[i][0],queries[i][1]
        ans[i]=vec[a]^vec[b+1]
    }
    return ans
}